tools/torchci/test_insights/ec2_pricing.py

```diff
@@ -0,0 +1,93 @@
+#!/usr/bin/env python3
+"""
+EC2 Pricing Map Generator
+
+Get pricing info for EC2 instances by reading .github/scale-config.yml and
+fetching current AWS pricing data.
+"""
+
+from functools import lru_cache
+from typing import Optional
+
+import requests
+import yaml
+
+
+@lru_cache
+def _get_scale_config() -> dict:
+    """Load scale-config.yml and return as a dictionary."""
+    with open(".github/scale-config.yml", "r") as f:
+        config = yaml.safe_load(f)
+    return config
+
+
+def get_ec2_instance_for_label(label: str) -> dict[str, Optional[str]]:
```

Comment:
Just a note that this wouldn't work with H100 and teams are using this nowadays to run tests, e.g. https://github.com/pytorch/pytorch/blob/main/.github/workflows/h100-symm-mem.yml.  Handling that is probably out of scope, but it's something we want to keep in mind I guess